### DISCUSSION TITLE
Using medusa modules in an existing nest project ?

### DISCUSSION THREAD

**Discussion by olup** [medusajs/medusa#5250] [Q&A]
My team would like to adopt medusa.js for the ecommerce part of our product (which is only one part) and expand the api so that we can benefit from a single user base, etc...
In the same time, we'd like to keep using industry standard frameworks like nest.js for the overall architecture.
Can medusa's modules be integrated in a nest project alongside other modules ? As I understand it medusa's rest api could be attached to any express server ?

Thanks.

**Comment by microCloudCode** [comment]
same question

**Comment by kenchoong** [comment]
same question as well

**Comment by BetterLuke** [comment]
same question as well

**Comment by kings-joy** [comment]
Is there a reason why Medusa is also not using NestJS?

**Comment by DhifMlak** [comment]
if you're using nestjs microservices you can simply use medusa as a microservice (which i think is the best approach) and link it

more info in this thread
[https://stackoverflow.com/questions/55942795/call-my-nestjs-microservice-with-nodejs-app](https://stackoverflow.com/questions/55942795/call-my-nestjs-microservice-with-nodejs-app)
